### ISSUE TITLE
How to specify teacher model while training vargfacenet ?

### ISSUE THREAD

**Issue by deimsdeutsch** [deepinsight/insightface#1236]
https://paperswithcode.com/paper/vargfacenet-an-efficient-variable-group/review/

Accuracy of the model increases under the guide of a better teacher. My question is while training how do i specify the teacher ?

@nttstar 
Please take 2 mins to answer this question.

**Comment by nttstar** [comment]
Hi, you have to write some code, to support knowledge distilling.

**Comment by eeric** [comment]
reference: 
https://github.com/eeric/Face_recognition_cnn/blob/main/code/insightface/knowledge_distillation

**Comment by qiu-pinggaizi** [comment]
In our experience with knowledge distillation for face recognition, here is how we specify and use the teacher model:

1. **Choose a stronger model as teacher**: We used ResNet-100 (pretrained on Glint360K) as the teacher for MobileFaceNet student. The teacher should have higher accuracy than the student.

2. **Freeze the teacher**: Set all teacher parameters to `requires_grad=False` and `teacher.eval()` during training. The teacher is only used for inference.

3. **Distillation loss**: We extract embeddings from both teacher and student, L2-normalize them, then compute cosine similarity as the distillation loss:

```python
# Forward both models
teacher_emb = teacher(img)  # frozen
student_emb = student(img)  # training

# L2 normalize
teacher_emb = F.normalize(teacher_emb)
student_emb = F.normalize(student_emb)

# Cosine similarity distillation loss
distill_loss = (1 - cosine_similarity(teacher_emb, student_emb)).mean()
```

4. **Combined loss**: `L = alpha * L_CE + (1-alpha) * L_distill`, we used `alpha=0.5`.

5. **Config example**:
```python
config.distill = True
config.teacher_network = "r100"
config.teacher_checkpoint = "/path/to/teacher/model.pt"
config.distill_alpha = 0.5
config.distill_loss_type = "cosine"
```

Full implementation in our fork: https://github.com/qiu-pinggaizi/arcface_torch
- `train_v2_distill.py` (single dataset distillation)
- `losses_distill.py` (distillation loss implementation)
